apps/meteor/server/services/upload/service.ts

```diff
@@ -0,0 +1,36 @@
+import type { IMessage, IUpload } from '@rocket.chat/core-typings';
+import { Meteor } from 'meteor/meteor';
+
+import { ServiceClassInternal } from '../../sdk/types/ServiceClass';
+import type {
+	ISendFileLivechatMessageParams,
+	ISendFileMessageParams,
+	IUploadFileParams,
+	IUploadService,
+} from '../../sdk/types/IUploadService';
+import { FileUpload } from '../../../app/file-upload/server';
+
+export class UploadService extends ServiceClassInternal implements IUploadService {
+	protected name = 'upload';
+
+	async uploadFile({ buffer, details, userId }: IUploadFileParams): Promise<IUpload> {
+		let uploadedFile: IUpload = {} as IUpload;
+		Meteor.runAsUser(userId, () => {
+			const fileStore = FileUpload.getStore('Uploads');
+			uploadedFile = fileStore.insertSync(details, buffer);
+		});
+		return uploadedFile;
```

Comment:
this doesn't need to be a `runAsUser`.. the confusion might be because `insertSync` requires a Fiber, but just regular `insert` can be used as well

```suggestion
		const fileStore = FileUpload.getStore('Uploads');
		return await fileStore.insert(details, buffer);
```